src/js/core/directives/ui-grid.js

```diff
@@ -239,32 +234,35 @@ function uiGridDirective($compile, $templateCache, $timeout, $window, gridUtil,
             if ($elm[0].offsetWidth <= 0 && sizeChecks < maxSizeChecks) {
               setTimeout(checkSize, sizeCheckInterval);
               sizeChecks++;
-            }
-            else {
-              $timeout(init);
+            } else {
+              $scope.$applyAsync(init);
```

Comment:
This should prevent an unnecessary digest cycle by pig backing on the next available cycle.